src/ORM/Marshaller.php

```diff
@@ -74,10 +75,18 @@ protected function _buildPropertyMap($options)
                 $key = $nested;
                 $nested = [];
             }
+            if ($key === '_joinData') {
+                continue;
+            }
             $assoc = $this->_table->association($key);
-            if ($assoc) {
-                $map[$assoc->property()] = ['association' => $assoc] + $nested + ['associated' => []];
+            if (!$assoc) {
+                throw new RuntimeException(sprintf(
```

Comment:
How is it a breaking change? The code could previously throw a RuntimeException. Not under these conditions, but it _could_ happen.